### PR TITLE
[BACK-2474] Add support for clinic membership restrictions

### PR DESCRIPTION
**Pull request by toddkazakov**

### CODE REVIEW

File: api/hydrophoneApi.go [hunk 1]
```diff
@@ -433,6 +434,54 @@ func (a *Api) ensureIdSet(ctx context.Context, userId string, confirmations []*m
...
+			// If a clinic has no restrictions the invite can be accepted
+			conf.Restrictions = &models.Restrictions{
+				CanAccept: true,
+			}
+			if resp.JSON200 != nil && resp.JSON200.Restrictions != nil && len(*resp.JSON200.Restrictions) > 0 {
+				// The clinic has configured restrictions, the user must match at least one
+				conf.Restrictions.CanAccept = false
+
+				for _, restriction := range *resp.JSON200.Restrictions {
+					if strings.HasSuffix(user.Username, fmt.Sprintf("@%s", *restriction.EmailDomain)) {
```
Comment (lostlevels) [inline api/hydrophoneApi.go:458]: I'm guessing there's no issues with case sensitivity here?

**Review by lostlevels — DISMISSED** [review]

File: api/hydrophoneApi.go (hunk 1 above)
Comment (toddkazakov) [reply api/hydrophoneApi.go:458]: Actually, this comparison should be case insensitive. Great catch!

**Review by lostlevels — APPROVED** [review]

**Review by lostlevels — APPROVED** [review]

**Pull request merged**